src/librustc_infer/infer/nll_relate/mod.rs

```diff
@@ -279,12 +273,7 @@ where
                 var
             }
 
-            _ => {
-                let projection = ty::ProjectionPredicate { projection_ty, ty: value_ty };
-                self.delegate
-                    .push_domain_goal(DomainGoal::Holds(WhereClause::ProjectionEq(projection)));
-                value_ty
-            }
+            _ => value_ty,
```

Comment:
ok so maybe I r+'d a bit too hastily -- it seems like we ought to be assering that this path is unreachable, no?

Reply:
Hmm, I can check

Reply:
I added back the same assertion that would have been hit before.